### PR TITLE
[Draft] Test cases for service node pool

### PR DESCRIPTION
**Pull request by neo954**
These are test cases are for GitHub task #3634

### CODE REVIEW

**Comment by hu-weihua** [comment]
@neo954, please paste the UT result of these 5 new cases in this PR. Thanks.

**Comment by hu-weihua** [comment]
@neo954, This task has been lasted for 4 weeks. Could you please paste your UT result out ASAP? 
@robin2008 Could you help to review after @neo954  paste his UT result? thank you very much.

File: xCAT-test/autotest/testcase/snpool/set_disjointdhcps_in_sn_pool_tftp_sync_mode [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+check:rc==0
+
+##
+# Do testing
+##
+##
+
+cmd:chdef -t site disjointdhcps=1
+check:rc==0
+cmd:chdef -t node testnode1 servicenode=$$SN,$$CN
```
Comment (robin2008) [inline xCAT-test/autotest/testcase/snpool/set_disjointdhcps_in_sn_pool_tftp_sync_mode]: when test with disjoint mode, you need to set the servicenode=sn01 and then you can check no boot configuration generated on sn02

File: xCAT-test/autotest/testcase/snpool/set_disjointdhcps_in_sn_pool_tftp_sync_mode [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+cmd:chdef -t network 20_0_0_0-255_0_0_0 net=20.0.0.0 mask=255.0.0.0 dynamicrange=20.0.0.1-20.0.0.2 tftpserver=$$SN dhcpserver=$$SN
+check:rc==0
+cmd:nodeset testnode1 osimage=rhels7.99-ppc64le-install-compute
+#check:rc==0
+cmd:xdsh $$SN 'test -f /tftpboot/petitboot/testnode1'
+check:rc==0
+cmd:xdsh $$CN 'test -f /tftpboot/petitboot/testnode1'
+check:rc==0
+
+##
```
Comment (robin2008) [inline xCAT-test/autotest/testcase/snpool/set_disjointdhcps_in_sn_pool_tftp_sync_mode]: And no need to set sharedtftp=1 again as all cases here are for rsync mode.

File: xCAT-test/autotest/testcase/snpool/set_disjointdhcps_in_sn_pool_tftp_sync_mode [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+cmd:test -f /tftpboot/petitboot/testnode1
+check:rc!=0
+cmd:xdsh $$SN 'test -f /tftpboot/petitboot/testnode1'
+check:rc==0
+cmd:xdsh $$CN 'test -f /tftpboot/petitboot/testnode1'
+check:rc==0
+cmd:nodeset testnode1 offline
+check:rc==0
+# DHCP dynamic range
+cmd:chdef -t network 20_0_0_0-255_0_0_0 net=20.0.0.0 mask=255.0.0.0 dynamicrange=20.0.0.1-20.0.0.2 tftpserver=$$SN dhcpserver=$$SN
```
Comment (robin2008) [inline xCAT-test/autotest/testcase/snpool/set_disjointdhcps_in_sn_pool_tftp_sync_mode]: this is to test if servicenode=sn02 and set dhcpserver=sn01, we need to check if the boot configuration files are generated on both dedicate servicenode and the dhcpserver.

**Comment by robin2008** [comment]
@neo954 I've finished the review, please enhance the case to cover the issue there.

**Comment by neo954** [comment]
The output of the unit test of test case `set_xcatmaster_in_sn_pool_tftp_sync_mode.out`. The original output is around 51MB. I have to use the attachment.

[set_xcatmaster_in_sn_pool_tftp_sync_mode.out.gz](https://github.com/xcat2/xcat-core/files/1445685/set_xcatmaster_in_sn_pool_tftp_sync_mode.out.gz)

**Comment by neo954** [comment]
@robin2008, The code was revised, please review it when you have time. Thanks.

**Pull request merged**